src/modules/chats/components/chat/chat.ts

```diff
@@ -1,68 +1,295 @@
+
+  chatInfo: Block;
+  page: number;
+  perPage: number;
+  lastResponseChatId: number;
+  loadPage: EventListener;
+  lastDate: string;
+  lastSender: string;
+
   constructor(props: Props) {
+
+    const chatInfo = new ChatInfo({
+      chat: props.chat,
+      events: {
+        click: (e: Event) => {
+          e.stopPropagation();
+          const target: HTMLElement = e.target as HTMLElement;
+          const elem: HTMLElement = document.querySelector(target.tagName === 'IMG' ? '.chat-image' : '.chat-users-modal')!;
+          elem.classList.add('opened');
+          elem.parentElement!.classList.add('opened');
+        }
+      }
+    });
+
     super({
       tagName: 'div',
       classList: 'chat'
     }, '', props, {
-      deleteChat: [new IconButton({
-        img: 'delete.svg',
-        type: 'button',
-        classList: 'chat__contact__delete',
-        events: {
-          click: (e: Event) => {
-            e.stopPropagation();
-            const elem: HTMLElement = document.querySelector('.delete-chat')!;
-            elem.classList.add('opened');
-            elem.parentElement!.classList.add('opened');
+      chatInfo: [chatInfo],
+      deleteChat: [],
+      compose: [new Compose()],
+      messages: []
+    });
+
+    this.chatInfo = chatInfo;
+    this.perPage = 20;
+    this.page = 0;
+
+    this.loadPage = (): void => {
+      const dialog: HTMLElement = this.getContent().querySelector('.dialog')!;
+      const dialogContainer: HTMLElement = this.getContent().querySelector('.dialog__container')!;
+      if ((dialog.getBoundingClientRect().top - 100) < dialogContainer.getBoundingClientRect().top) {
+        this.props.socket.send(JSON.stringify({
+          content: (this.perPage * this.page).toString(),
+          type: 'get old',
+        }));
+        this.page++;
+      }
+    }
+  }
+
+  ping(socket: WebSocket): void {
+    const interval = setInterval(() => {
+      socket.send(JSON.stringify({ type: 'ping' }));
+    }, 30000);
+    socket.onclose = () => clearInterval(interval);
+  }
+
+  scrollBottom() {
+    const dialog: HTMLElement = this.getContent().querySelector('.dialog')!;
+    dialog.scrollTop = dialog.scrollHeight;
+  }
+
+  initChat(messages: Array<messageDTO>): void {
+    const messagesArray: Array<Props> = [];
+    for (let i = 0; i < messages.length; i++) {
+      const m: messageDTO = messages[i];
+      const nextMessage: messageDTO | undefined = messages[i + 1];
+      const dateFormattedNext = calcChatDate(nextMessage?.time);
+      const dateFormatted = calcChatDate(m.time);
+
+      const senderNext = nextMessage?.user_id;
+      const sender = m.user_id;
+
+      messagesArray.push({
+        message: m,
+        timeFormatted: getLocalTime(m.time),
+        user: this.props.user,
+        sender: sender === senderNext ? undefined : this.props.users.find((u: userDTO) => u.id === +m.user_id)
+      });
+      if (dateFormattedNext !== dateFormatted) messagesArray.push({ dateFormatted });
```

Comment:
Здесь тоже было бы более читаемо, если код будет внутри блока фигурных скобок
```suggestion
      if (dateFormattedNext !== dateFormatted) {
            messagesArray.push({ dateFormatted });
      }
```